server/v2/cometbft/server.go

```diff
@@ -68,6 +69,21 @@ func (s *CometBFTServer[T]) Init(appI serverv2.AppI[T], v *viper.Viper, logger l
 		AppTomlConfig:    appTomlConfig,
 	}
 
+	chainID := v.GetString(FlagChainID)
```

Comment:
That flag doesn't exist on the start command.
We should add it.

Reply:
Lead to an issue when flag & genesis chain id mismatch. Nodes should use the same chain-id, if not will cause appHash.
Also should add for both v1 & v2.

Reply:
You are right, it is probably only useful on query / txs commands. Nevermind this comment then 👍